src/mpid/ch4/src/ch4_init.c

```diff
@@ -414,16 +414,17 @@ int MPID_Init(int requested, int *provided)
 
     /* Initialize multiple VCIs */
     /* TODO: add checks to ensure MPIDI_vci_t is padded or aligned to MPL_CACHELINE_SIZE */
-    MPIDI_global.n_vcis = 1;
-    if (MPIR_CVAR_CH4_NUM_VCIS > 1) {
-        MPIDI_global.n_vcis = MPIR_CVAR_CH4_NUM_VCIS;
-        /* There are configured maxes that we need observe. */
-        /* TODO: check them at configure time to avoid discrepancy */
-        MPIR_Assert(MPIDI_global.n_vcis <= MPIDI_CH4_MAX_VCIS);
-        MPIR_Assert(MPIDI_global.n_vcis <= MPIR_REQUEST_NUM_POOLS);
-    }
+    MPIR_Assert(MPIR_CVAR_CH4_NUM_VCIS >= 1);   /* maximum number vcis can be reserved */
+    MPIR_Assert(MPIR_CVAR_CH4_RESERVE_VCIS >= 0);       /* maximum number vcis can be reserved */
```

Comment:
I updated the comment for `MPIR_CVAR_CH4_NUM_VCIS`. That was the original comment, but now we changed its meaning to number of implicit vcis rather than maximum number of vcis.